src/MediaStreamTrack.ts

```diff
@@ -107,7 +119,16 @@ export default class MediaStreamTrack extends EventTarget<MediaStreamTrackEventM
             throw new Error('Only implemented for video tracks');
         }
 
-        WebRTCModule.mediaStreamTrackSwitchCamera(this.id);
+        const constraints = deepClone(this._settings);
```

Comment:
Question: should we just set the facingMode and not any other setting, since different cameras might have different supported resolutions?

Reply:
Old behavior of switch camera kept the target resolution, and worst case it'll just find the closest available resolution so it should be fine until we can remove this entirely.

Reply:
Sounds good.